system/ee/ExpressionEngine/Addons/channel/libraries/channel_form/Channel_form_lib.php

```diff
@@ -2822,37 +2821,62 @@ public function initialize($reinitialize = false)
 
         $this->fetch_settings();
 
-        $this->option_fields = $this->native_option_fields;
+        if (empty($this->option_fields)) {
+
+            $cache_key = '/Fieldtype/OptionFieldtypes';
+            $this->option_fields = ee()->cache->get($cache_key);
+
+            if (empty($this->option_fields)) {
+                $fieldtypes = ee('Model')->get('Fieldtype')->all()->pluck('name');
+
+                ee()->load->library('api');
+                ee()->legacy_api->instantiate('channel_fields');
+
+                // Get the list of Fieldtypes that extend OptionFieldtype
+                $this->option_fields = array_filter($fieldtypes, function($fieldtype) {
+                    ee()->api_channel_fields->include_handler($fieldtype);
+                    $class = ucfirst($fieldtype) . '_ft';
+
+                    return is_subclass_of($class, 'OptionFieldtype');
+                });
+
+                $this->option_fields = array_values($this->option_fields);
+
+                ee()->cache->save($cache_key, $this->option_fields);
+            }
+        }
+
         /*
-                ee()->config->load('config');
+            TODO: I think the following code can be removed
```

Comment:
@gilbert-hop I don't think we can afford to remove this code without prior deprecation, as that might break existing functionality for the sites that are using those config overrides.

Reply:
I suggested just because the code block was commented out.

Reply:
Oh, you're right. Let's keep it as is then